credential/manifest/model.go

```diff
@@ -141,4 +141,80 @@ func (cf *CredentialResponse) IsValid() error {
 	return util.NewValidator().Struct(cf)
 }
 
+// IsValidPair validates the rules on how a credential manifest [cm] and credential application [ca] relate to each other https://identity.foundation/credential-manifest/#credential-application
+func IsValidPair(cm CredentialManifest, ca CredentialApplication) error {
+
+	err := cm.IsValid()
+	if err != nil {
+		return errors.Wrap(err, "credential manifest is not valid")
+	}
+
+	err = ca.IsValid()
+	if err != nil {
+		return errors.Wrap(err, "credential application is not valid")
+	}
+
+	// The object MUST contain a manifest_id property. The value of this property MUST be the id of a valid Credential Manifest.
+	if cm.ID != ca.ManifestID {
+		return errors.New("the credential application's manifest id must be equal to the credential manifest's id")
+	}
+
+	// The ca must have a format property if the related Credential Manifest specifies a format property.
+	// Its value must be a subset of the format property in the Credential Manifest that this Credential Submission
+	if !cm.Format.IsEmpty() {
+		cmFormats := map[string]bool{}
+		for _, format := range cm.Format.FormatValues() {
+			cmFormats[format] = true
+		}
+
+		for _, format := range ca.Format.FormatValues() {
+			if cmFormats[format] == false {
```

Comment:
would this ever be false? may be best as a check like...

`if _, ok := cmFormats[format]; !ok { .... `